pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -102,7 +103,11 @@ func (a *Actuator) Create(ctx context.Context, machine *machinev1.Machine) error
 		var detailedError autorest.DetailedError
 		if errors.As(err, &detailedError) {
 			statusCode, ok := detailedError.StatusCode.(int)
-			if ok && statusCode >= 400 && statusCode < 500 {
+			// Any 4xx error that isn't invalid credentials should be a terminal failure.
+			// Invalid Credentials implies that the credentials expired between the scope creation and API calls,
+			// this may happen when CCO is refreshing credentials simultaneously.
+			// In this case we should retry as the credentials should be updated in the secret.
+			if ok && statusCode >= 400 && statusCode < 500 && !azure.InvalidCredentials(err) {
```

Comment:
can we just do `&& statusCode != 401` without creating `InvalidCredentials` function?

Reply:
or we can have one function that handles all status codes

Reply:
We could inline it, but I was thinking having an explicit function that is InvalidCredentials makes this more readable. Ie I can see that we are including 400 < 500 errors, but not ones that are invalid credentials. Excluding 401 you then have to know what that status code means